dry-types: add some basic examples to intro page

georgemillo opened the pull request:
Add some quick and dirty code examples so that someone coming newly to the gem can get an overview of what the gem is about, and its basic usage.

solnic commented on source/gems/dry-types/index.html.md:
```diff
@@ -16,7 +16,80 @@ sections:
+See [Built-In Types](/gems/dry-types/built-in-types/) for a full list of types.
+
+You can use dry-types with [dry-struct](http://dry-rb.org/gems/dry-struct)
+to create a class whose attributes have clearly-defined, coercible types:
```
Maybe just "to create a class with typed attributes"? They don't have to be coercible, or even strict, it depends on the use cases.

georgemillo replied:
fair point. Fixed ;)

georgemillo commented:
Just rewrote this for clarity with more info based on our conversation [here](https://github.com/dry-rb/dry-types/issues/163), and added a new example about `meta`.

flash-gordon commented:
@solnic is it good to go? This is the last one :)

solnic commented:
Sorry, I wasn't watching this project and missed this one
